tcp_protocol/src/lib.rs

```diff
@@ -1,15 +1,55 @@
+//! Kanata TCP Protocol
+//!
+//! This crate defines the JSON message format for communication between
+//! TCP clients and the Kanata keyboard remapping daemon.
+
 use serde::{Deserialize, Serialize};
 use std::str::FromStr;
 
+/// Messages sent from the server to connected clients.
 #[derive(Debug, Serialize, Deserialize)]
 pub enum ServerMessage {
-    LayerChange { new: String },
-    LayerNames { names: Vec<String> },
-    CurrentLayerInfo { name: String, cfg_text: String },
-    ConfigFileReload { new: String },
-    CurrentLayerName { name: String },
-    MessagePush { message: serde_json::Value },
-    Error { msg: String },
+    LayerChange {
+        new: String,
+    },
+    LayerNames {
+        names: Vec<String>,
+    },
+    CurrentLayerInfo {
+        name: String,
+        cfg_text: String,
+    },
+    ConfigFileReload {
+        new: String,
+    },
+    CurrentLayerName {
+        name: String,
+    },
+    MessagePush {
+        message: serde_json::Value,
+    },
+    Error {
+        msg: String,
+    },
+    /// Response to `Hello` command with server capabilities.
+    /// Introduced in protocol v1.11.
+    HelloOk {
+        version: String,
+        protocol: u8,
+        capabilities: Vec<String>,
+    },
+    /// Basic status info response.
+    /// Introduced in protocol v1.11.
+    StatusInfo {
+        ready: bool,
+    },
+    /// Response to Reload commands when `wait: true` was specified.
+    /// Introduced in protocol v1.11.
+    ReloadResult {
+        ready: bool,
```

Comment:
The `ready` naming in the protocol is inaccurate according to what it means in the underlying server. Naming should reflect "was config loading successful".